.github/workflows/actions.yml

```diff
@@ -0,0 +1,109 @@
+name: CI
+
+on:
+  push:
+    branches: '**'
+    tags: 'v*'
+  pull_request:
+    branches: '**'
+
+env:
+  BUILD_VERSION: ${{ github.ref_name }}.${{ github.run_number }}
+
+jobs:
+  build-and-test:
+    runs-on: windows-2022
+
+    steps:
+      - name: Cut v from release tag and set version string
+        if: startsWith(github.ref, 'refs/tags/v')
+        shell: pwsh
+        run: |
+          if (-not ($env:BUILD_VERSION -match "^v((\d+).(\d+).(\d+).(\d+))$")) {
+            Write-Host "Invalid version number: $env:BUILD_VERSION"
+            exit 1
+          }
+
+          echo "BUILD_VERSION=$($matches[1])" >> $env:GITHUB_ENV
+
+      - name: Set version string to 1.0.0.0 for non-release
+        if: "!startsWith(github.ref, 'refs/tags/v')"
```

Comment:
FYI quotes necessary as otherwise syntax is incorrect :grin: (https://github.com/orgs/community/discussions/25017)